src/json.hpp

```diff
@@ -2818,22 +2820,21 @@ class lexer
     std::size_t chars_read = 0;
     /// the start position of the current token
     std::size_t start_pos = 0;
-
     /// buffer for variable-length tokens (numbers, strings)
-    std::vector<char> yytext = std::vector<char>(1024, '\0');
+    std::vector<value_t> yytext = std::vector<value_t>(1024, '\0');
```

Comment:
Much of the rest of the code expects `yytext` to be `std::vector<char>`, such as line 2658 where a `char` is added to it, or line 2568 where `yytext.data()` is passed to `std::strtoull`.